src/ripple/app/main/Application.cpp

```diff
@@ -354,13 +354,31 @@ class ApplicationImp
 
     static
     std::size_t
-    numberOfThreads(Config const& config)
+    asioThreadCount(Config const& config)
     {
-    #if RIPPLE_SINGLE_IO_SERVICE_THREAD
+#if RIPPLE_SINGLE_IO_SERVICE_THREAD
         return 1;
-    #else
+#else
         return (config.NODE_SIZE >= 2) ? 2 : 1;
-    #endif
+#endif
+    }
+
+    // Calculate the number of threads for the job queue
+    // taking into account the configured node size for
+    // this instance.
+    static
+    unsigned int
+    jobThreadCount (Config const& config)
+    {
+        if (config.standalone())
+            return 1;
+
+        unsigned int const minThreads = 2;
+        unsigned int const maxThreads = 2 + (config.NODE_SIZE * 4);
+
+        return 2 + std::max(
```

Comment:
The addition of 2 could be removed and built into the bounds, simplifying a bit.
```
unsigned int const minThreads = 4;
unsigned int const maxThreads = minThreads + (config.NODE_SIZE * 4);
```